flict/flictlib/license.py

```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
-            logger.license_logger.debug(
-                "LEAF, returning " + str(expanded_list))
-            logger.license_logger.debug(
-                "cop: " + interim_license_expression_list)
-            #print("managed____ \""  + str(expanded_list) + "\"  <---- MIDDLE")
             return expanded_list
 
         current_op = interim_license_expression_list.op
-        logger.license_logger.debug("cop: " + current_op)
         for lep in interim_license_expression_list.list:
-            logger.license_logger.debug(" ------ lep ----- " + str(lep))
             if current_op is None:
                 raise Exception("Internal failure. No operator found")
 
-            elif current_op == "OR":
-                lep_list = self.interim_license_expression_set_list(lep)
+            lep_list = self.interim_license_expression_set_list(lep)
+            if current_op == "OR":
                 expanded_list = self._manage_list_item_or(
                     expanded_list, lep_list)
 
             elif current_op == "AND":
-                lep_list = self.interim_license_expression_set_list(lep)
                 expanded_list = self._manage_list_item_and(
                     expanded_list, lep_list)
-        #print("managed____ \""  + str(expanded_list) + "\"  <---- FINAL")
         return expanded_list
 
     def _manage_list_item_and(self, license_list, lep):
-        logger.license_logger.debug(" * Andy")
         if isinstance(lep, LicenseExpressionList):
-            raise Exception("Internal failure. Wrong type " +
-                            str(type(lep)) + " for: " + str(lep))
-        else:
-            # single license
-            if len(license_list) == 0:
-                logger.license_logger.debug("wooops: " + str(license_list))
-                logger.license_logger.debug("wooops: " + str(lep))
-                license_list = lep
-            else:
-                logger.license_logger.debug("daisy: " + str(license_list))
-                logger.license_logger.debug("daisy: " + str(lep))
-                logger.license_logger.debug(
-                    " -------------====== Andy ====----------------- SINGLE: " + str(license_list))
-                new_list = []
-                for item in license_list:
-                    logger.license_logger.debug(
-                        "  item: " + str(item) + " <--- " + str(lep))
-                    for lep_item in lep:
-                        logger.license_logger.debug(
-                            "    item: " + str(item) + " <--- " + str(lep_item))
-                        new_item = list(set(item + lep_item))
-                        logger.license_logger.debug("    item: " + str(item))
-                        new_list.append(new_item)
-                    logger.license_logger.debug("    list: " + str(new_list))
-                license_list = new_list
-            return license_list
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
```

Comment:
Shouldn't that be a FlictError instead?
I honestly don't want to introduce plain Exceptions where it's not needed

Reply:
Good point. Was thinking about that myself when I crossed this.

Reply:
Fixed